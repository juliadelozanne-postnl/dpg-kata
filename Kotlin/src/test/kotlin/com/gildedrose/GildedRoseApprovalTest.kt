package com.gildedrose

import org.approvaltests.combinations.CombinationApprovals
import org.junit.jupiter.api.Test

internal class GildedRoseApprovalTest {

    @Test
    fun `combination approval tests for aged brie items`() {
        fun updateQualityItem(sellIn: Int, quality: Int): String {
            val items = listOf(AgedBrie(sellIn, quality))
            val app = GildedRose(items)
            app.updateItems()
            return items[0].toString()
        }

        val sellIns = arrayOf(0, 6, 11, 12)
        val qualities = arrayOf(0, 10, 49, 50)

        CombinationApprovals.verifyAllCombinations(::updateQualityItem, sellIns, qualities)
    }

    @Test
    fun `combination approval tests for conjured items`() {
        fun updateQualityItem(sellIn: Int, quality: Int): String {
            val items = listOf(ConjuredItem(sellIn, quality))
            val app = GildedRose(items)
            app.updateItems()
            return items[0].toString()
        }

        val sellIns = arrayOf(0, 6, 11, 12)
        val qualities = arrayOf(0, 10, 49, 50)

        CombinationApprovals.verifyAllCombinations(::updateQualityItem, sellIns, qualities)
    }

    /*@Test
    fun `update quality approval test`() {
        fun updateQualityItem(name: String, sellIn: Int, quality: Int): String {
            val items = listOf(Item(name, sellIn, quality))
            val app = GildedRose(items)
            app.updateQuality()
            return items[0].toString()
        }

        val sellIns = arrayOf(0, 6, 11, 12)
        val qualities = arrayOf(0, 10, 49, 50)
        val names = arrayOf(
            "Sulfuras, Hand of Ragnaros",
            "Aged Brie",
            "Backstage passes to a TAFKAL80ETC concert",
            "+5 Dexterity Vest",
            "Elixir of the Mongoose"
        )

        CombinationApprovals.verifyAllCombinations(::updateQualityItem, names, sellIns, qualities)
    }*/
}
