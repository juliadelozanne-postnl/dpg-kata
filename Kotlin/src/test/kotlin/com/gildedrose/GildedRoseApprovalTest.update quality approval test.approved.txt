[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0
[Sulfuras, Hand of Ragnaros, 6, 10] => Sulfuras, Hand of Ragnaros, 6, 10
[Sulfuras, Hand of Ragnaros, 6, 49] => Sulfuras, Hand of Ragnaros, 6, 49
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0
[Sulfuras, Hand of Ragnaros, 11, 10] => Sulfuras, Hand of Ragnaros, 11, 10
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50
[Sulfuras, Hand of Ragnaros, 12, 0] => Sulfuras, Hand of Ragnaros, 12, 0
[Sulfuras, Hand of Ragnaros, 12, 10] => Sulfuras, Hand of Ragnaros, 12, 10
[Sulfuras, Hand of Ragnaros, 12, 49] => Sulfuras, Hand of Ragnaros, 12, 49
[Sulfuras, Hand of Ragnaros, 12, 50] => Sulfuras, Hand of Ragnaros, 12, 50
[Aged Brie, 0, 0] => Aged Brie, -1, 2
[Aged Brie, 0, 10] => Aged Brie, -1, 12
[Aged Brie, 0, 49] => Aged Brie, -1, 50
[Aged Brie, 0, 50] => Aged Brie, -1, 50
[Aged Brie, 6, 0] => Aged Brie, 5, 1
[Aged Brie, 6, 10] => Aged Brie, 5, 11
[Aged Brie, 6, 49] => Aged Brie, 5, 50
[Aged Brie, 6, 50] => Aged Brie, 5, 50
[Aged Brie, 11, 0] => Aged Brie, 10, 1
[Aged Brie, 11, 10] => Aged Brie, 10, 11
[Aged Brie, 11, 49] => Aged Brie, 10, 50
[Aged Brie, 11, 50] => Aged Brie, 10, 50
[Aged Brie, 12, 0] => Aged Brie, 11, 1
[Aged Brie, 12, 10] => Aged Brie, 11, 11
[Aged Brie, 12, 49] => Aged Brie, 11, 50
[Aged Brie, 12, 50] => Aged Brie, 11, 50
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2
[Backstage passes to a TAFKAL80ETC concert, 6, 10] => Backstage passes to a TAFKAL80ETC concert, 5, 12
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1
[Backstage passes to a TAFKAL80ETC concert, 11, 10] => Backstage passes to a TAFKAL80ETC concert, 10, 11
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1
[Backstage passes to a TAFKAL80ETC concert, 12, 10] => Backstage passes to a TAFKAL80ETC concert, 11, 11
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Backstage passes to a TAFKAL80ETC concert, 11, 50
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Backstage passes to a TAFKAL80ETC concert, 11, 50
[+5 Dexterity Vest, 0, 0] => +5 Dexterity Vest, -1, 0
[+5 Dexterity Vest, 0, 10] => +5 Dexterity Vest, -1, 8
[+5 Dexterity Vest, 0, 49] => +5 Dexterity Vest, -1, 47
[+5 Dexterity Vest, 0, 50] => +5 Dexterity Vest, -1, 48
[+5 Dexterity Vest, 6, 0] => +5 Dexterity Vest, 5, 0
[+5 Dexterity Vest, 6, 10] => +5 Dexterity Vest, 5, 9
[+5 Dexterity Vest, 6, 49] => +5 Dexterity Vest, 5, 48
[+5 Dexterity Vest, 6, 50] => +5 Dexterity Vest, 5, 49
[+5 Dexterity Vest, 11, 0] => +5 Dexterity Vest, 10, 0
[+5 Dexterity Vest, 11, 10] => +5 Dexterity Vest, 10, 9
[+5 Dexterity Vest, 11, 49] => +5 Dexterity Vest, 10, 48
[+5 Dexterity Vest, 11, 50] => +5 Dexterity Vest, 10, 49
[+5 Dexterity Vest, 12, 0] => +5 Dexterity Vest, 11, 0
[+5 Dexterity Vest, 12, 10] => +5 Dexterity Vest, 11, 9
[+5 Dexterity Vest, 12, 49] => +5 Dexterity Vest, 11, 48
[+5 Dexterity Vest, 12, 50] => +5 Dexterity Vest, 11, 49
[Elixir of the Mongoose, 0, 0] => Elixir of the Mongoose, -1, 0
[Elixir of the Mongoose, 0, 10] => Elixir of the Mongoose, -1, 8
[Elixir of the Mongoose, 0, 49] => Elixir of the Mongoose, -1, 47
[Elixir of the Mongoose, 0, 50] => Elixir of the Mongoose, -1, 48
[Elixir of the Mongoose, 6, 0] => Elixir of the Mongoose, 5, 0
[Elixir of the Mongoose, 6, 10] => Elixir of the Mongoose, 5, 9
[Elixir of the Mongoose, 6, 49] => Elixir of the Mongoose, 5, 48
[Elixir of the Mongoose, 6, 50] => Elixir of the Mongoose, 5, 49
[Elixir of the Mongoose, 11, 0] => Elixir of the Mongoose, 10, 0
[Elixir of the Mongoose, 11, 10] => Elixir of the Mongoose, 10, 9
[Elixir of the Mongoose, 11, 49] => Elixir of the Mongoose, 10, 48
[Elixir of the Mongoose, 11, 50] => Elixir of the Mongoose, 10, 49
[Elixir of the Mongoose, 12, 0] => Elixir of the Mongoose, 11, 0
[Elixir of the Mongoose, 12, 10] => Elixir of the Mongoose, 11, 9
[Elixir of the Mongoose, 12, 49] => Elixir of the Mongoose, 11, 48
[Elixir of the Mongoose, 12, 50] => Elixir of the Mongoose, 11, 49
